temporalio/worker/_activity.py

```diff
@@ -214,6 +214,9 @@ def _cancel(
             warnings.warn(f"Cannot find activity to cancel for token {task_token!r}")
             return
         logger.debug("Cancelling activity %s, reason: %s", task_token, cancel.reason)
+        activity.cancellation_details.set_details(
+            temporalio.activity.ActivityCancellationDetails._from_proto(cancel.details)
+        )
         activity.cancel(cancelled_by_request=True)
```

Comment:
```suggestion
        activity.cancel(cancelled_by_request=cancel.details.is_cancelled)
```

I think we can only set `cancelled_by_request` as `True` when the server actually cancels. We use this value to indicate whether we can send activity cancel back to server. The server won't accept cancel in any other situation, only if cancel is requested server side.

For instance, I believe that if the activity was paused and not canceled, the server may error when trying to send cancel back saying "cancel was never requested". May want to have a test case confirming.

Reply:
Yes I think you're right:
```
2025-05-05T18:08:00.561239Z  WARN temporal_sdk_core::worker::activities: Network error while completing activity error=Status { code: InvalidArgument, message: "unable to mark activity as canceled without activity being request canceled first", metadata: MetadataMap { headers: {"content-type": "application/grpc"} }, source: None }
```

which I get if I re-throw the activity error in the test:
```
@activity.defn
def sync_heartbeat_activity() -> (
    Optional[temporalio.activity.ActivityCancellationDetails]
):
    while True:
        try:
            activity.heartbeat()
            time.sleep(1)
        except (CancelledError, asyncio.CancelledError):
            raise asyncio.CancelledError()
```

changed to `is_cancelled`